roachtest: update cancel/tpcc to use long-running queries

SHA: https://github.com/cockroachdb/cockroach/commits/4c52530a33367c58e434111324fcda8c3d73582a

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cancel/tpcc/distsql/w=10,nodes=3 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1568299&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20191101-1568299/cancel/tpcc/distsql/w=10_nodes=3/run_1
	cancel.go:65: query completed before it could be canceled

```

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1724655&tab=buildLog) on [master@e49530d6e7c9b48c47329c89da8533deccd964bf](https://github.com/cockroachdb/cockroach/commits/e49530d6e7c9b48c47329c89da8533deccd964bf):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1724655-1580726149-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:734: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:734
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1724655&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1725352&tab=buildLog) on [master@4ec5929a08687b82b05ef80231fddd0d53421973](https://github.com/cockroachdb/cockroach/commits/4ec5929a08687b82b05ef80231fddd0d53421973):

```
		    main.(*cluster).RunE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2000
		    main.(*cluster).Run
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:1933
		    main.registerCancel.func1.1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:45
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2344
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1725352-1580762820-19-n3cpu4:1 -- ./workload fixtures load tpcc --warehouses=10 {pgurl:1} returned:
		    stderr:
		    I200203 21:08:42.420081 1 ccl/workloadccl/cliccl/fixtures.go:279  starting restore of 9 tables
		    
		    stdout::
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func3
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2416
		    main.(*monitor).wait.func4
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2445
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor command failure:
		  - signal: broken pipe

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1725352&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1732360&tab=buildLog) on [master@dbeb04dae240e93222a87dd6aea89db4cd3735cc](https://github.com/cockroachdb/cockroach/commits/dbeb04dae240e93222a87dd6aea89db4cd3735cc):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1732360-1580975954-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1732360&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1745452&tab=buildLog) on [master@de7a7bb804716835579e713ab7264f8af168a14f](https://github.com/cockroachdb/cockroach/commits/de7a7bb804716835579e713ab7264f8af168a14f):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1745452-1581495003-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1745452&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1747374&tab=buildLog) on [master@2739821b911d777fa2a927295d699b559360a802](https://github.com/cockroachdb/cockroach/commits/2739821b911d777fa2a927295d699b559360a802):

```
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1747374-1581581035-19-n3cpu4:1 -- ./workload fixtures load tpcc --warehouses=10 {pgurl:1} returned:
		    stderr:
		    I200213 08:20:40.160831 1 ccl/workloadccl/cliccl/fixtures.go:279  starting restore of 9 tables
		    Error: restoring fixture: sql: expected 6 destination arguments in Scan, not 7
		    Error:  exit status 1
		    
		    stdout::
		  - exit status 1

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1747374&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1749385&tab=buildLog) on [master@e671a4ef97cbc6cf5d22f8f322fd45733d302094](https://github.com/cockroachdb/cockroach/commits/e671a4ef97cbc6cf5d22f8f322fd45733d302094):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1749385-1581667748-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1749385&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1761088&tab=buildLog) on [master@a51ddb8342ff54283baf5c7556cb1d4ff8c4e5da](https://github.com/cockroachdb/cockroach/commits/a51ddb8342ff54283baf5c7556cb1d4ff8c4e5da):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1761088-1582358300-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1761088&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1764605&tab=buildLog) on [master@abe944fb0ad990ceaba7e592cc83b8922639016c](https://github.com/cockroachdb/cockroach/commits/abe944fb0ad990ceaba7e592cc83b8922639016c):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1764605-1582617752-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1764605&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1766891&tab=buildLog) on [master@8b5adba703fae9b6961623f65b685d93b0fe0290](https://github.com/cockroachdb/cockroach/commits/8b5adba703fae9b6961623f65b685d93b0fe0290):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1766891-1582705160-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1766891&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1783198&tab=buildLog) on [master@d6a52a8f6f07b3b6b32af2243075e5365fb21c45](https://github.com/cockroachdb/cockroach/commits/d6a52a8f6f07b3b6b32af2243075e5365fb21c45):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1783198-1583307906-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1783198&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1801614&tab=buildLog) on [master@5570c01402796edb7cd06eb8ce7f615371f22d42](https://github.com/cockroachdb/cockroach/commits/5570c01402796edb7cd06eb8ce7f615371f22d42):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1801614-1583911542-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:747: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:747
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1801614&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1808695&tab=buildLog) on [master@33d71472dc01cbc5064b3c5e1fcd666a33f606de](https://github.com/cockroachdb/cockroach/commits/33d71472dc01cbc5064b3c5e1fcd666a33f606de):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1808695-1584256913-20-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1808695&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

@asubiotto @yuzefovich 

```
panic: runtime error: slice bounds out of range [1:0]

goroutine 6745 [running]:
panic(0x40bb280, 0xc004eec700)
	/usr/local/go/src/runtime/panic.go:722 +0x2c2 fp=0xc00b019960 sp=0xc00b0198d0 pc=0x79a432
runtime.goPanicSliceB(0x1, 0x0)
	/usr/local/go/src/runtime/panic.go:103 +0xa3 fp=0xc00b0199a8 sp=0xc00b019960 pc=0x798f73
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).DrainHelper(0xc000771800, 0xaaaaaaaaaaaaaaaa)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:633 +0x327 fp=0xc00b019a20 sp=0xc00b0199a8 pc=0x20c5327
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc000771800, 0x7773e6, 0xc000e182b0, 0xc00b019b40, 0x7773e6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:704 +0x17f fp=0xc00b019a88 sp=0xc00b019a20 pc=0x20ddd5f
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).DrainHelper(0xc003ff3800, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:631 +0xa0 fp=0xc00b019b00 sp=0xc00b019a88 pc=0x20c50a0
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).DrainMeta(0xc003ff3800, 0x4b86060, 0xc003a1dd10, 0x2, 0x3, 0xc003e15200)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:142 +0x5b fp=0xc00b019bb8 sp=0xc00b019b00 pc=0x22c379b
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).sendMetadata(0xc005bfb3b0, 0x4b86060, 0xc003a1dd10, 0x7fae257360b0, 0xc0039bc920, 0x4b01540, 0xc003986280, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:268 +0xcf fp=0xc00b019dd0 sp=0xc00b019bb8 pc=0x2b42edf
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).runWithStream(0xc005bfb3b0, 0x4b86060, 0xc003a1dd10, 0x7fae257360b0, 0xc0039bc920, 0xc0039bc5a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:310 +0x182 fp=0xc00b019e48 sp=0xc00b019dd0 pc=0x2b43472
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).Run(0xc005bfb3b0, 0x4b85fa0, 0xc0039a8c40, 0x4afdc20, 0xc0003d6780, 0xbd17e66700000001, 0x1036bda5b64e661c, 0xf335adcb, 0x6, 0xc0039bc5a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:167 +0x586 fp=0xc00b019f18 sp=0xc00b019e48 pc=0x2b425c6
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupRemoteOutputStream.func1(0x4b85fa0, 0xc0039a8c40, 0xc0039bc5a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:549 +0x9a fp=0xc00b019f90 sp=0xc00b019f18 pc=0x2b4d07a
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreatorHelper).accumulateAsyncComponent.func1.1(0xc00425b760, 0x4b85fa0, 0xc0039a8c40, 0xc0039bc5a0, 0xc003aab630)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1096 +0x44 fp=0xc00b019fb8 sp=0xc00b019f90 pc=0x2b4d434
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc00b019fc0 sp=0xc00b019fb8 pc=0x7caaa1
created by github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreatorHelper).accumulateAsyncComponent.func1
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1095 +0x6f
```

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1826302&tab=buildLog) on [master@6c3c5d4a9a0b837538a558fa22dcd4ff712fd8f5](https://github.com/cockroachdb/cockroach/commits/6c3c5d4a9a0b837538a558fa22dcd4ff712fd8f5):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1826302-1585121307-19-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1826302&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #44481 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202001281357_v19.2.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202001281357_v19.2.3) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1853455&tab=buildLog) on [master@b1a0b989bbfef500075a485edc762fe42ca7b32a](https://github.com/cockroachdb/cockroach/commits/b1a0b989bbfef500075a485edc762fe42ca7b32a):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1853455-1586071414-20-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2368,cancel.go:100,cancel.go:119,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2309
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1853455&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #46776 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).cancel/tpcc/distsql/w=10,nodes=3 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1877809&tab=buildLog) on [master@8c928666de3f90cf17d81d70a3ee759c39e2ce27](https://github.com/cockroachdb/cockroach/commits/8c928666de3f90cf17d81d70a3ee759c39e2ce27):

```
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1877809-1586945944-21-n3cpu4:1 -- ./cockroach sql --insecure -e "CANCEL QUERIES
		    	SELECT query_id FROM [SHOW CLUSTER QUERIES] WHERE query not like '%SHOW CLUSTER QUERIES%'" returned:
		    stderr:
		    
		    stdout:
		    CANCEL QUERIES 0:
		  - context canceled

	cluster.go:2378,cancel.go:100,cancel.go:119,test_runner.go:753: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2366
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2374
		    main.registerCancel.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:100
		    main.registerCancel.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cancel.go:119
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2422
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - error with attached stack trace:
		    main.init
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2319
		    runtime.doInit
		    	/usr/local/go/src/runtime/proc.go:5222
		    runtime.main
		    	/usr/local/go/src/runtime/proc.go:190
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - Goexit() was called

```

<details><summary>More</summary><p>


Artifacts: [/cancel/tpcc/distsql/w=10,nodes=3](https://teamcity.cockroachdb.com/viewLog.html?buildId=1877809&tab=artifacts#/cancel/tpcc/distsql/w=10,nodes=3)
Related:
- #46776 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44277 roachtest: cancel/tpcc/distsql/w=10,nodes=3 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Acancel%2Ftpcc%2Fdistsql%2Fw%3D10%2Cnodes%3D3.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cmd/roachtest/BUILD.bazel
+++ b/pkg/cmd/roachtest/BUILD.bazel
@@ -142,6 +142,7 @@ go_library(
         "//pkg/ts/tspb",
         "//pkg/util",
         "//pkg/util/binfetcher",
+        "//pkg/util/cancelchecker",
         "//pkg/util/contextutil",
         "//pkg/util/ctxgroup",
         "//pkg/util/httputil",

--- a/pkg/cmd/roachtest/cancel.go
+++ b/pkg/cmd/roachtest/cancel.go
@@ -113,7 +113,7 @@ func registerCancel(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:    fmt.Sprintf("cancel/tpcc/distsql/w=%d,nodes=%d", warehouses, numNodes),
-		Owner:   OwnerSQLExec,
+		Owner:   OwnerSQLQueries,
 		Cluster: makeClusterSpec(numNodes),
 		Run: func(ctx context.Context, t *test, c *cluster) {
 			runCancel(ctx, t, c, queries, warehouses, true /* useDistsql */)
@@ -122,7 +122,7 @@ func registerCancel(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:    fmt.Sprintf("cancel/tpcc/local/w=%d,nodes=%d", warehouses, numNodes),
-		Owner:   OwnerSQLExec,
+		Owner:   OwnerSQLQueries,
 		Cluster: makeClusterSpec(numNodes),
 		Run: func(ctx context.Context, t *test, c *cluster) {
 			runCancel(ctx, t, c, queries, warehouses, false /* useDistsql */)

--- a/pkg/cmd/roachtest/cancel.go
+++ b/pkg/cmd/roachtest/cancel.go
@@ -15,7 +15,10 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/util/cancelchecker"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/cockroach/pkg/workload/tpch"
+	"github.com/cockroachdb/errors"
 )
 
 // Motivation:
@@ -23,7 +26,7 @@ import (
 // insufficient in detecting problems with canceling long-running, multi-node
 // DistSQL queries. This is because, unlike local queries which only need to
 // cancel the transaction's context, DistSQL queries must cancel flow contexts
-// on each node involed in the query. Typical strategies for local execution
+// on each node involved in the query. Typical strategies for local execution
 // testing involve using a builtin like generate_series to create artificially
 // long-running queries, but these approaches don't create multi-node DistSQL
 // queries; the only way to do so is by querying a large dataset split across
@@ -33,41 +36,43 @@ import (
 // Once DistSQL queries provide more testing knobs, these tests can likely be
 // replaced with unit tests.
 func registerCancel(r *testRegistry) {
-	runCancel := func(ctx context.Context, t *test, c *cluster,
-		queries []string, warehouses int, useDistsql bool) {
-		t.Skip("skipping flaky cancel/tpcc test", "test needs to be updated see https://github.com/cockroachdb/cockroach/issues/42103")
+	runCancel := func(ctx context.Context, t *test, c *cluster, tpchQueriesToRun []int, useDistsql bool) {
 		c.Put(ctx, cockroach, "./cockroach", c.All())
-		c.Put(ctx, workload, "./workload", c.All())
 		c.Start(ctx, t, c.All())
 
 		m := newMonitor(ctx, c, c.All())
 		m.Go(func(ctx context.Context) error {
-			t.Status("importing TPCC fixture")
-			c.Run(ctx, c.Node(1), tpccImportCmd(warehouses))
+			t.Status("restoring TPCH dataset for Scale Factor 1")
+			if err := loadTPCHDataset(ctx, t, c, 1 /* sf */, newMonitor(ctx, c), c.All()); err != nil {
+				t.Fatal(err)
+			}
 
 			conn := c.Conn(ctx, 1)
 			defer conn.Close()
 
-			var queryPrefix string
+			queryPrefix := "USE tpch; "
 			if !useDistsql {
-				queryPrefix = "SET distsql = off;"
+				queryPrefix += "SET distsql = off; "
 			}
 
 			t.Status("running queries to cancel")
-			for _, q := range queries {
+			for _, queryNum := range tpchQueriesToRun {
 				sem := make(chan struct{}, 1)
-				go func(q string) {
-					t.l.Printf("executing \"%s\"\n", q)
+				go func(query string) {
+					t.l.Printf("executing \"%s\"\n", query)
 					sem <- struct{}{}
-					_, err := conn.Exec(queryPrefix + q)
+					_, err := conn.Exec(queryPrefix + query)
 					if err == nil {
 						close(sem)
 						t.Fatal("query completed before it could be canceled")
 					} else {
 						fmt.Printf("query failed with error: %s\n", err)
+						if !errors.Is(err, cancelchecker.QueryCanceledError) {
+							t.Fatal("unexpected error")
+						}
 					}
 					sem <- struct{}{}
-				}(q)
+				}(tpch.QueriesByNumber[queryNum])
 
 				<-sem
 
@@ -88,7 +93,7 @@ func registerCancel(r *testRegistry) {
 						t.Fatal("query could not be canceled")
 					}
 					timeToCancel := timeutil.Now().Sub(cancelStartTime)
-					fmt.Printf("canceling \"%s\" took %s\n", q, timeToCancel)
+					fmt.Printf("canceling q%d took %s\n", queryNum, timeToCancel)
 
 				case <-time.After(5 * time.Second):
 					t.Fatal("query took too long to respond to cancellation")
@@ -100,32 +105,26 @@ func registerCancel(r *testRegistry) {
 		m.Wait()
 	}
 
-	const warehouses = 10
 	const numNodes = 3
-	queries := []string{
-		`SELECT * FROM tpcc.stock`,
-		`SELECT * FROM tpcc.stock WHERE s_quantity > 100`,
-		`SELECT s_i_id, sum(s_quantity) FROM tpcc.stock GROUP BY s_i_id`,
-		`SELECT * FROM tpcc.stock ORDER BY s_quantity`,
-		`SELECT * FROM tpcc.order_line JOIN tpcc.stock ON s_i_id=ol_i_id`,
-		`SELECT ol_number, sum(s_quantity) FROM tpcc.stock JOIN tpcc.order_line ON s_i_id=ol_i_id WHERE ol_number > 10 GROUP BY ol_number ORDER BY ol_number`,
-	}
+	// Choose several longer running TPCH queries (each is taking at least 3s to
+	// complete).
+	tpchQueriesToRun := []int{7, 9, 20, 21}
 
 	r.Add(testSpec{
-		Name:    fmt.Sprintf("cancel/tpcc/distsql/w=%d,nodes=%d", warehouses, numNodes),
+		Name:    fmt.Sprintf("cancel/tpch/distsql/queries=%v,nodes=%d", tpchQueriesToRun, numNodes),
 		Owner:   OwnerSQLQueries,
 		Cluster: makeClusterSpec(numNodes),
 		Run: func(ctx context.Context, t *test, c *cluster) {
-			runCancel(ctx, t, c, queries, warehouses, true /* useDistsql */)
+			runCancel(ctx, t, c, tpchQueriesToRun, true /* useDistsql */)
 		},
 	})
 
 	r.Add(testSpec{
-		Name:    fmt.Sprintf("cancel/tpcc/local/w=%d,nodes=%d", warehouses, numNodes),
+		Name:    fmt.Sprintf("cancel/tpch/local/queries=%v,nodes=%d", tpchQueriesToRun, numNodes),
 		Owner:   OwnerSQLQueries,
 		Cluster: makeClusterSpec(numNodes),
 		Run: func(ctx context.Context, t *test, c *cluster) {
-			runCancel(ctx, t, c, queries, warehouses, false /* useDistsql */)
+			runCancel(ctx, t, c, tpchQueriesToRun, false /* useDistsql */)
 		},
 	})
 }

--- a/pkg/cmd/roachtest/scrub.go
+++ b/pkg/cmd/roachtest/scrub.go
@@ -48,7 +48,7 @@ func makeScrubTPCCTest(
 
 	return testSpec{
 		Name:    fmt.Sprintf("scrub/%s/tpcc/w=%d", optionName, warehouses),
-		Owner:   OwnerSQLExec,
+		Owner:   OwnerSQLQueries,
 		Cluster: makeClusterSpec(numNodes),
 		Run: func(ctx context.Context, t *test, c *cluster) {
 			runTPCC(ctx, t, c, tpccOptions{

--- a/pkg/cmd/roachtest/sqlsmith.go
+++ b/pkg/cmd/roachtest/sqlsmith.go
@@ -254,7 +254,7 @@ func registerSQLSmith(r *testRegistry) {
 		r.Add(testSpec{
 			Name: fmt.Sprintf("sqlsmith/setup=%s/setting=%s", setup, setting),
 			// NB: sqlsmith failures should never block a release.
-			Owner:      OwnerSQLExec,
+			Owner:      OwnerSQLQueries,
 			Cluster:    makeClusterSpec(4),
 			MinVersion: "v20.2.0",
 			Timeout:    time.Minute * 20,

--- a/pkg/cmd/roachtest/test_registry.go
+++ b/pkg/cmd/roachtest/test_registry.go
@@ -34,7 +34,7 @@ const (
 	OwnerCDC           Owner = `cdc`
 	OwnerKV            Owner = `kv`
 	OwnerPartitioning  Owner = `partitioning`
-	OwnerSQLExec       Owner = `sql-exec`
+	OwnerSQLQueries    Owner = `sql-queries`
 	OwnerSQLSchema     Owner = `sql-schema`
 	OwnerStorage       Owner = `storage`
 )
@@ -81,9 +81,8 @@ var roachtestOwners = map[Owner]OwnerMetadata{
 	OwnerPartitioning: {SlackRoom: `kv`, Mention: []string{`@cockroachdb/kv`},
 		TriageColumnID: 3550674,
 	},
-	// TODO(jordanlewis): that team doesn't exist any more, fix things up.
-	OwnerSQLExec: {SlackRoom: `sql-execution-team`, Mention: []string{`@cockroachdb/sql-queries`},
-		TriageColumnID: 6837155,
+	OwnerSQLQueries: {SlackRoom: `sql-queries`, Mention: []string{`@cockroachdb/sql-queries`},
+		TriageColumnID: 13549252,
 	},
 	OwnerSQLSchema: {SlackRoom: `sql-schema`, Mention: []string{`@cockroachdb/sql-schema`},
 		TriageColumnID: 8946818,

--- a/pkg/cmd/roachtest/tpcc.go
+++ b/pkg/cmd/roachtest/tpcc.go
@@ -379,7 +379,7 @@ func registerTPCC(r *testRegistry) {
 	})
 	r.Add(testSpec{
 		Name:       "tpcc/interleaved/nodes=3/cpu=16/w=500",
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		MinVersion: "v20.1.0",
 		Cluster:    makeClusterSpec(4, cpu(16)),
 		Timeout:    6 * time.Hour,

--- a/pkg/cmd/roachtest/tpcdsvec.go
+++ b/pkg/cmd/roachtest/tpcdsvec.go
@@ -167,7 +167,7 @@ func registerTPCDSVec(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:       "tpcdsvec",
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		Cluster:    makeClusterSpec(3),
 		MinVersion: "v20.1.0",
 		Run: func(ctx context.Context, t *test, c *cluster) {

--- a/pkg/cmd/roachtest/tpchbench.go
+++ b/pkg/cmd/roachtest/tpchbench.go
@@ -164,7 +164,7 @@ func registerTPCHBenchSpec(r *testRegistry, b tpchBenchSpec) {
 
 	r.Add(testSpec{
 		Name:       strings.Join(nameParts, "/"),
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		Cluster:    makeClusterSpec(numNodes),
 		MinVersion: minVersion,
 		Run: func(ctx context.Context, t *test, c *cluster) {

--- a/pkg/cmd/roachtest/tpchvec.go
+++ b/pkg/cmd/roachtest/tpchvec.go
@@ -646,7 +646,7 @@ const tpchVecNodeCount = 3
 func registerTPCHVec(r *testRegistry) {
 	r.Add(testSpec{
 		Name:       "tpchvec/perf",
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		Cluster:    makeClusterSpec(tpchVecNodeCount),
 		MinVersion: "v19.2.0",
 		Run: func(ctx context.Context, t *test, c *cluster) {
@@ -656,7 +656,7 @@ func registerTPCHVec(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:    "tpchvec/disk",
-		Owner:   OwnerSQLExec,
+		Owner:   OwnerSQLQueries,
 		Cluster: makeClusterSpec(tpchVecNodeCount),
 		// 19.2 version doesn't have disk spilling nor memory monitoring, so
 		// there is no point in running this config on that version.
@@ -668,7 +668,7 @@ func registerTPCHVec(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:       "tpchvec/smithcmp",
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		Cluster:    makeClusterSpec(tpchVecNodeCount),
 		MinVersion: "v20.1.0",
 		Run: func(ctx context.Context, t *test, c *cluster) {
@@ -678,7 +678,7 @@ func registerTPCHVec(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:       "tpchvec/perf_no_stats",
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		Cluster:    makeClusterSpec(tpchVecNodeCount),
 		MinVersion: "v20.2.0",
 		Run: func(ctx context.Context, t *test, c *cluster) {
@@ -688,7 +688,7 @@ func registerTPCHVec(r *testRegistry) {
 
 	r.Add(testSpec{
 		Name:       "tpchvec/bench",
-		Owner:      OwnerSQLExec,
+		Owner:      OwnerSQLQueries,
 		Cluster:    makeClusterSpec(tpchVecNodeCount),
 		MinVersion: "v20.2.0",
 		Skip: "This config can be used to perform some benchmarking and is not " +

